lighty-modules/lighty-aaa-aggregator/lighty-aaa/src/main/java/io/lighty/aaa/AAALighty.java

```diff
@@ -10,60 +10,197 @@
+                "org.opendaylight.aaa.impl.provider");
+        mainHandler.addServlet(idmLightServlet, "/*");
+        server.addContextHandler(contexts);
+        this.handlers.add(contexts);
+        this.handlers.add(mainHandler);
+        this.shiroWebEnvironmentLoaderListener = new ShiroWebEnvironmentLoaderListener(shiroConfiguration,
+                dataBroker,
+                certificateManager,
+                authenticationService,
+                tokenAuthenticators,
+                tokenStore,
+                passwordHashService);
+        server.addCommonEventListener(shiroWebEnvironmentLoaderListener);
 
-        AAALightyShiroProvider aaaLightyShiroProvider;
+        final Map<String, String> properties = new HashMap<>();
+        final CustomFilterAdapterConfigurationImpl customFilterAdapterConfig =
+                new CustomFilterAdapterConfigurationImpl();
+        customFilterAdapterConfig.update(properties);
+        final FilterHolder customFilterAdapter = new FilterHolder(new CustomFilterAdapter(customFilterAdapterConfig));
+        server.addCommonFilter(customFilterAdapter, "/*");
 
-        void setAaaLightyShiroProvider(final AAALightyShiroProvider aaaLightyShiroProvider) {
-            this.aaaLightyShiroProvider = aaaLightyShiroProvider;
+        final FilterHolder shiroFilter = new FilterHolder(new AAAShiroFilter());
+        server.addCommonFilter(shiroFilter, "/*");
+
+        final FilterHolder crossOriginFilter = new FilterHolder(new CrossOriginFilter());
+        crossOriginFilter.setInitParameter("allowedMethods", "GET,POST,OPTIONS,DELETE,PUT,HEAD");
+        crossOriginFilter.setInitParameter("allowedHeaders",
+                "origin, content-type, accept, authorization, Authorization");
+        server.addCommonFilter(crossOriginFilter, "/*");
+    }
+
+    @Override
+    protected boolean initProcedure() throws InterruptedException {
+        this.passwordHashService = defaultPasswordHashService;
+        if (credentialAuth == null) {
+            IdmLightProxy idmLightProxy = new IdmLightProxy(iidmStore, defaultPasswordHashService);
+            this.credentialAuth = idmLightProxy;
+            this.claimCache = idmLightProxy;
         }
+        this.tokenAuthenticators = buildTokenAuthenticators(this.credentialAuth);
+        try {
+            final StoreBuilder storeBuilder = new StoreBuilder(iidmStore);
+            final String domain = storeBuilder.initDomainAndRolesWithoutUsers(IIDMStore.DEFAULT_DOMAIN);
+            if (domain != null) {
+                // If is not exist. If domain already exist on path, will be used instead
+                storeBuilder.createUser(domain, aaaConfiguration.getUsername(), aaaConfiguration.getPassword(), true);
+            }
 
-        AAALightyShiroProvider getAaaLightyShiroProvider() {
-            return this.aaaLightyShiroProvider;
+        } catch (final IDMStoreException e) {
+            LOG.error("Failed to initialize data in store", e);
```

Comment:
Add `return false;` to this exception

Reply:
adding `return false ` would mean that other handlers after would not get a chance to close. Also im using lambda `.forEach` expression which is void. I fixed this issue using `AtomicBoolean`